test/common.py

```diff
@@ -2567,52 +2609,65 @@ def browser_open(cls, url):
       # Recreate the new data directory.
       os.mkdir(browser_data_dir)
 
-      if is_chrome():
-        config = ChromeConfig()
-      elif is_firefox():
-        config = FirefoxConfig()
-      else:
-        exit_with_error(f'EMTEST_BROWSER_AUTO_CONFIG only currently works with firefox or chrome. EMTEST_BROWSER was "{EMTEST_BROWSER}"')
+      if not config:
+        exit_with_error(f'EMTEST_BROWSER_AUTO_CONFIG only currently works with firefox, chrome and safari. EMTEST_BROWSER was "{EMTEST_BROWSER}"')
       if WINDOWS:
         # Escape directory delimiter backslashes for shlex.split.
         browser_data_dir = browser_data_dir.replace('\\', '\\\\')
       config.configure(browser_data_dir)
       browser_args += f' {config.data_dir_flag}"{browser_data_dir}"'
 
     browser_args = shlex.split(browser_args)
+    if is_safari():
+      # For the macOS 'open' command, pass
+      #   --new: to make a new Safari app be launched, rather than add a tab to an existing Safari process/window
+      #   --fresh: do not restore old tabs (e.g. if user had old navigated windows open)
+      #   --background: Open the new Safari window behind the current Terminal window, to make following the test run more pleasing (this is for convenience only)
+      #   -a <exe_name>: The path to the executable to open, in this case Safari
+      browser_args = ['open', '--new', '--fresh', '--background', '-a'] + browser_args
```

Comment:
Shouldn't these args go in `SafariConfig`?

Reply:
Or maybe these should be used also for Chrome and FF on macOS?    i.e. is `open` the way we should be launching all browsers on macOS?